halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/deployment/v1/DistributedDeployer.java

```diff
@@ -276,16 +276,44 @@ public RemoteAction deploy(DistributedServiceProvider<T> serviceProvider,
     List<Integer> allRoscos = new ArrayList<>(executionsByServerGroupVersion.keySet());
     allRoscos.sort(Integer::compareTo);
 
-    int roscoCount = allRoscos.size();
-    if (roscoCount <= MAX_REMAINING_SERVER_GROUPS) {
+    cleanupServerGroups(details, roscoService, roscoSettings, executionsByServerGroupVersion, allRoscos);
+  }
+
+  private <S, T extends Account> void cleanupServerGroups(AccountDeploymentDetails<T> details,
+      DistributedService<S, T> service,
+      ServiceSettings settings,
+      Map<Integer, Integer> workloadByVersion,
+      List<Integer> runningVersions) {
+    int runningVersionCount = runningVersions.size();
+
+    if (runningVersionCount <= 1) {
```

Comment:
This happens on every fresh deploy - you'll only have one of each service. I'll add a log statement but it's expected to be hit